main.rb

```diff
@@ -0,0 +1,141 @@
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    array = to_a
+    i = 0
+    while i < array.length
+      yield(array[i])
+      i += 1
+    end
+    array
+  end
```

Comment:
Enumerable `#my_each` when a block is given when self is a range returns the range itself after calling the given block once for each element in self

    Failure/Error: expect(range.my_each(&block)).to eq(range.each(&block))

       expected: `5..50`
            got: `[5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50]`

       (compared using ==)
     # ./enumerables_spec.rb:56:in `block (5 levels) in <top (required)>'